### PR TITLE
Type Grape

### PR DESCRIPTION
**Pull request by AlexEvanczuk**
Basic types for https://github.com/ruby-grape/grape

Some discussion about this here:
https://sorbet-ruby.slack.com/archives/CHN2L03NH/p1563314645490800

> Hi everyone, I’ve been discussing an interesting thing around typing gems and code in general with @Jeff Carbonella.  I am typing this gem in `sorbet-typed`: https://github.com/ruby-grape/grape.
> 
> What we’re finding is this gem uses a lot of fairly complex, somewhat hard to follow meta-programming techniques to get methods into the `Grape::API` class that all of our own application API code inherits from.
> 
> Originally, I was trying to figure out all of the implementation details of how the methods got into there, and tried to type the gem based on implementation details.
> 
> Later, we realized that we can simply type the methods in `Grape::API` and forget about the implementation details entirely.  We felt this was better because:
> A) Types should serve as documentation and hide implementation detail
> B) The inside implementation of a gem should be considered private API, and we should only type public API
> C) Public API types are kind of like an “integration test” over a “unit test.”  We don’t really need to unit test our code — we just want to make sure the available API works as expected (this is similar to B).
> 
> We welcome any thoughts on this and are curious if there are any opposing opinions on it! (edited)

### CODE REVIEW

**Comment by DarkDimius** [comment]
Thank you!

File: lib/grape/all/grape.rbi [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+  sig do
+    params(
+      format: Symbol,
+    ).void
+  end
+  def self.format(format)
+  end
+end
+
```
Comment (DarkDimius) [inline lib/grape/all/grape.rbi]:
```suggestion
```

**Comment by DarkDimius** [comment]
Thank you!

File: lib/grape/all/grape.rbi [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      )
+    ).void
+  end
+  def self.mount(api_class)
+  end
+
+  sig do
+    params(
+      object: T.untyped,
+      with: T.nilable(T.class_of(Grape::Entity))
```
Comment (ghiculescu) [inline lib/grape/all/grape.rbi:190]: i think this line will crash if you don't have the https://github.com/ruby-grape/grape-entity gem installed
